lib/src/Uncertainty/Algorithm/Simulation/NAIS.cxx

```diff
@@ -92,17 +92,17 @@ Distribution NAIS::computeAuxiliaryDistribution(const Sample & sample,
   Collection<Distribution> margins(dimensionSample);
 
   // Computation of auxiliary distribution using ot.Mixture
-  for (UnsignedInteger k = 0; k < dimensionSample ; ++k)
+  const UnsignedInteger numberOfSample = getMaximumOuterSampling() * getBlockSize();
+  Collection<Distribution> collectionOfDistribution(numberOfSample);
+  for (UnsignedInteger i = 0; i < numberOfSample ; ++i)
   {
-    const UnsignedInteger numberOfSample = getMaximumOuterSampling() * getBlockSize();
-    Collection<Distribution> collectionOfDistribution(numberOfSample);
-    for (UnsignedInteger i = 0; i < numberOfSample ; ++i)
-    {
-      collectionOfDistribution[i] = Normal(sample(i, k), silverman[k]);
-    }
-    margins[k] = Mixture(collectionOfDistribution, weights);
-  } // for k
-  return ComposedDistribution(margins);
+    const Point  mean_normal =  sample[i];
```

Comment:
For complex types (such as Point) use the constructor. Also use camelCase for the variable names:
``` 
const Point meanNormal(sample[i]);
```

Reply:
Thank you, modification has been done accordingly